components/resourceDetails/resourceSharing/sharingAccordion/index.test.jsx

```diff
@@ -20,19 +20,107 @@
  */
 
 import React from "react";
+import userEvent from "@testing-library/user-event";
+import { waitFor, within } from "@testing-library/react";
 import { renderWithTheme } from "../../../../__testUtils/withTheme";
+import mockPersonContact from "../../../../__testUtils/mockPersonContact";
+import mockAddressBook from "../../../../__testUtils/mockAddressBook";
 import SharingAccordion from "./index";
 import { TESTCAFE_ID_AGENT_ACCESS_TABLE } from "../agentAccessTable";
+import { TESTCAFE_ID_ADVANCED_SHARING_BUTTON } from "../advancedSharingButton";
+import useContacts from "../../../../src/hooks/useContacts";
+import useAddressBook from "../../../../src/hooks/useAddressBook";
+import { TESTCAFE_ID_CUSTOM_POLICY_DROPDOWN } from "../customPolicyDropdown";
+
+jest.mock("../../../../src/hooks/useContacts");
+const mockedUseContacts = useContacts;
+
+jest.mock("../../../../src/hooks/useAddressBook");
+const mockedUseAddressBook = useAddressBook;
 
 describe("SharingAccordion", () => {
-  afterEach(() => {
-    jest.restoreAllMocks();
+  beforeEach(() => {
+    mockedUseAddressBook.mockReturnValue({ data: mockAddressBook() });
+    mockedUseContacts.mockReturnValue({
+      data: [
+        mockPersonContact(
```

Comment:
Something I like to do to encourage reuse of mocked variables, is to set them as constants outside of the top describe scope. It also makes the uses of them a bit more easy to read IMO. Not needed here, just wanted to tip about it.